sbtc/src/idpack/tests.rs

```diff
@@ -0,0 +1,183 @@
+        let mut current = base;
+
+        for inc in increments {
+            current += inc;
+            if current <= MAX_ID_VALUE {
+                values.push(current);
+            }
+        }
+
+        prop_assume!(!values.is_empty());
+        roundtrip_test(&values).expect("round-trip failed");
+    }
+
+    #[test]
+    fn test_roundtrip_mixed_density(
+        dense_runs in prop::collection::vec((1..100u64, 1..20usize), 1..10),
+        gaps in prop::collection::vec(1..10000u64, 1..10)
+    ) {
+        // Create sequences with mixed density patterns
+        let mut values = Vec::new();
+        let mut current = 1u64;
+
+        for (i, (step, count)) in dense_runs.into_iter().enumerate() {
+            // Add a gap before each dense run (except first)
+            if i > 0 && i - 1 < gaps.len() {
+                current += gaps[i - 1];
+            }
+
+            // Add a dense run of values
+            for _ in 0..count {
+                if current <= MAX_ID_VALUE {
+                    values.push(current);
+                    current += step;
+                }
+            }
+        }
+
+        let sorted_unique = to_sorted_unique(&values);
+        prop_assume!(!sorted_unique.is_empty());
+        roundtrip_test(&sorted_unique).expect("round-trip failed");
+    }
+
+    #[test]
+    fn test_roundtrip_edge_values(
+        small_values in prop::collection::vec(1..100u64, 1..50),
+        large_values in prop::collection::vec((MAX_ID_VALUE - 10000)..MAX_ID_VALUE, 1..50)
+    ) {
+        // Combine small and large values
+        let mut values = small_values;
+        values.extend(large_values);
+
+        let sorted_unique = to_sorted_unique(&values);
+        prop_assume!(!sorted_unique.is_empty());
+        roundtrip_test(&sorted_unique).expect("round-trip failed");
+    }
+}
+
+/// Helper function to ensure test data is sorted and unique
+fn to_sorted_unique(values: &[u64]) -> Vec<u64> {
+    let mut set = BTreeSet::new();
+    set.extend(values);
+    set.into_iter().collect()
+}
+
+/// Performs the full round-trip test: package -> encode -> decode -> compare
+fn roundtrip_test(values: &[u64]) -> Result<(), String> {
+    // Skip empty sets (handled by prop_assume in the test functions)
+    if values.is_empty() {
+        return Ok(());
+    }
+
+    // Step 1: Package the values into segments
+    let segmenter = BitmapSegmenter;
+    let segments = segmenter.package(values).expect("segmentation failed");
+
+    // Step 2: Encode the segments to bytes
+    let encoded_bytes = segments.encode();
+
+    // Step 3: Decode the bytes back to segments
+    let decoded_segments = Segments::decode(&encoded_bytes).expect("decoding failed");
+
+    // Step 4: Extract values from decoded segments
+    let decoded_values = decoded_segments.values().collect::<Vec<_>>();
+
+    // Step 5: Sort and deduplicate (in case there are any overlaps between segments)
+    let decoded_unique = to_sorted_unique(&decoded_values);
```

Comment:
Shouldn't this never be necessary?

Reply:
Yeah you're right, removed in [add7e07](https://github.com/stacks-network/sbtc/pull/1525/commits/add7e07533111382bb762b3125ee94f89fa9a194)